tests/artifacts/test_artifacts.py

```diff
@@ -0,0 +1,71 @@
+import pathlib
+
+import pytest
+
+import mlflow
+from mlflow.exceptions import MlflowException
+
+
+@pytest.fixture()
+def run_with_artifact(tmp_path):
+    artifact_path = "test"
+    artifact_content = "content"
+    local_path = tmp_path.joinpath("file.txt")
+    local_path.write_text(artifact_content)
+    with mlflow.start_run() as run:
+        mlflow.log_artifact(local_path, artifact_path)
+
+    return (run, artifact_path, artifact_content)
+
+
+def test_download_artifacts_with_uri(run_with_artifact):
+    run, artifact_path, artifact_content = run_with_artifact
+    run_uri = f"runs:/{run.info.run_id}/{artifact_path}"
+    actual_uri = pathlib.PurePosixPath(run.info.artifact_uri) / artifact_path
+    for uri in (run_uri, actual_uri):
+        download_output_path = mlflow.artifacts.download_artifacts(artifact_uri=uri)
+        downloaded_artifact_path = list(pathlib.Path(download_output_path).iterdir())[0]
+        assert downloaded_artifact_path.read_text() == artifact_content
+
+
+def test_download_artifacts_with_run_id_and_path(run_with_artifact):
+    run, artifact_path, artifact_content = run_with_artifact
+    download_output_path = mlflow.artifacts.download_artifacts(
+        run_id=run.info.run_id, artifact_path=artifact_path
+    )
+    downloaded_artifact_path = list(pathlib.Path(download_output_path).iterdir())[0]
+    assert downloaded_artifact_path.read_text() == artifact_content
+
+
+def test_download_artifacts_with_run_id_no_path(run_with_artifact):
+    run, artifact_path, _ = run_with_artifact
+    artifact_relative_path_top_level_dir = pathlib.Path(artifact_path).parts[0]
```

Comment:
```suggestion
    artifact_relative_path_top_level_dir = pathlib.PurePosixPath(artifact_path).parts[0]
```

Probably not a big issue, but should this be `PurePosixPath` because `artifact_path` is a posix path?

Reply:
From what I understand, the "Pure" variant is to limit the access of code to not permit write capabilities to the underlying file system. For reference access to paths, Path and the Pure variants behave identically. I suppose this is a design consideration (do we want to be explicit in path manipulations that don't create directories, write files, remove files, etc) to enforce the 'Pure' variants when reading or constructing representations of paths and only use the abstract top-level Path class when interfacing with the OS? (Interested to hear everyone's thoughts on this one)

Reply:
Good call, @harupy ! This should be PurePosixPath because MLflow artifact paths always use the POSIX-style construction (`/` separator). Updated!